Sandbox-sensitive test

**Describe the bug**
A clear and concise description of what the bug is.

This part of the pytest test `TestUIUIFileDialog.test_process_event`:

https://github.com/MyreMylar/pygame_gui/blob/cc1c41da88f0556012ace7256c897f07aa8b39e9/tests/test_windows/test_ui_file_dialog.py#L399-L404

fails with 

```
>       assert Path(file_dialog.current_directory_path).name == ''
E       AssertionError: assert 'images' == ''
E         
E         + images
```

when ran in a nix sandbox on macos. I think it assumes `/` has the `x` permission?

uncovered in https://github.com/NixOS/nixpkgs/pull/352150#issuecomment-2445832850

**To Reproduce**
Steps to reproduce the behaviour:

`nix build --system aarch64-darwin github:nixos/nixpkgs/refs/pull/352150/head#python311Packages.pygame-gui --check`

**Expected behaviour**
build success

**Platform and software (please complete the following information):**
 - OS: nixos 24.05
 - Pygame GUI version: 0.6.2
 - Pygame version: 2.5.2 CE edition (also reproduced on 2.5.1)

--- a/tests/test_windows/test_ui_file_dialog.py
+++ b/tests/test_windows/test_ui_file_dialog.py
@@ -1,3 +1,4 @@
+import platform
 from pathlib import Path
 
 import pygame
@@ -396,12 +397,13 @@ class TestUIUIFileDialog:
 
         assert Path(file_dialog.current_directory_path).name == 'images'
 
-        file_dialog.file_path_text_line.set_text('/')
+        if platform.system().upper() == 'WINDOWS':
+            file_dialog.file_path_text_line.set_text('/')
 
-        file_dialog.process_event(pygame.event.Event(UI_TEXT_ENTRY_FINISHED,
-                                                     {'ui_element': file_dialog.file_path_text_line}))
+            file_dialog.process_event(pygame.event.Event(UI_TEXT_ENTRY_FINISHED,
+                                                         {'ui_element': file_dialog.file_path_text_line}))
 
-        assert Path(file_dialog.current_directory_path).name == ''
+            assert Path(file_dialog.current_directory_path).name == ''
 
     def test_show(self, _init_pygame, default_ui_manager, _display_surface_return_none):
         file_dialog = UIFileDialog(rect=pygame.Rect(100, 100, 440, 500),

